src/MAUI/Maui.Samples/Samples/Layers/AddCustomDynamicEntityDataSource/SimulatedDataSource.cs

```diff
@@ -0,0 +1,187 @@
+﻿using Esri.ArcGISRuntime.Data;
+using Esri.ArcGISRuntime.Geometry;
+using Esri.ArcGISRuntime.RealTime;
+using System.Diagnostics;
+using System.Text.Json;
+
+
+namespace ArcGIS.Samples.AddCustomDynamicEntityDataSource
+{
+    public class SimulatedDataSource : DynamicEntityDataSource
+    {
+        // Hold a reference to the file stream reader, the process task, and the cancellation token source.
+        private Task? _processTask;
+        private StreamReader? _streamReader;
+        private CancellationTokenSource? _cancellationTokenSource;
+        private List<Field> _fields;
+
+        public SimulatedDataSource(string filePath, string entityIdField, TimeSpan delay)
+        {
+            FilePath = filePath;
+            EntityIdField = entityIdField;
+            Delay = delay;
+        }
+
+        #region Properties
+
+        // Expose the file path, entity ID field, and delay length as properties.
+        public string FilePath { get; }
+        public string EntityIdField { get; }
+        public TimeSpan Delay { get; }
+        public SpatialReference SpatialReference { get; set; } = SpatialReferences.Wgs84;
```

Comment:
To simplify, we can probably remove this property since we're assuming WGS84 for this data source.